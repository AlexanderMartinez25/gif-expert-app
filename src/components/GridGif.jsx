import { useEffect, useState } from "react";
import { getGif } from "../helpers/getGifs";

export const GridGif = ({ category }) => {
  const [counter, setCounter] = useState(10);
  // efectos secundarios, si le pasamos [] como segundo argumento
  // solo se dispararé una única vez
  useEffect(() => {
    getGif(category);
  }, []);

  return (
    <>
      <h3>{category}</h3>

      <h5>{counter}</h5>
      <button onClick={() => setCounter(counter + 1)}>+1</button>
    </>
  );
};
