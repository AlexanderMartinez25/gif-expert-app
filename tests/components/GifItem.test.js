import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas a <GifItem/>', () => {
  const title = 'Hola Mundo';
  const url = 'https://youtube.com/img.png';

  test('debe evualuar Snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el tÃ­tulo en el componente', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy;
  });
});
