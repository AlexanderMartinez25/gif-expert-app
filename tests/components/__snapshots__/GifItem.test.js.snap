// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifItem/> debe evualuar Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola Mundo"
      src="https://youtube.com/img.png"
    />
    <p>
      Hola Mundo
    </p>
  </div>
</div>
`;
